src/engine/runtime.js

```diff
@@ -1963,10 +1971,15 @@ class Runtime extends EventEmitter {
      * @param {!Target} editingTarget New editing target.
      */
     setEditingTarget (editingTarget) {
+        const oldEditingTarget = this._editingTarget;
```

Comment:
It might make sense to just skip everything and early-out if the editing target isn't really changing

Reply:
Hmmm that's a good point but I'm not sure I want to make that change here, since it seems like more of a refactor that could have unintentional side effects rather than necessary for this PR.